Source/WebKit/Platform/IPC/haiku/ConnectionHaiku.cpp

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) 2019 Haiku, Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+#include "config.h"
+#include "Connection.h"
+
+namespace IPC{
+    void Connection::platformInitialize(Identifier identifier)
+    {
+    	fprintf(stderr,"%s\n",__PRETTY_FUNCTION__);
+    	m_connectedProcess = identifier;
+    }
+    void Connection::platformInvalidate()
+    {
+
+    }
+    void Connection::runReadEventLoop()
+    {
+
+    }
+    void Connection::runWriteEventLoop()
+    {
+
+    }
+    bool Connection::open()
+    {
+    	status_t result = m_messenger.SetTo(NULL,m_connectedProcess);
+    	if(result == B_OK)
+    	{
+    		fprintf(stderr,"%s\n",__PRETTY_FUNCTION__);
+    		return true;
+    	}
+    	
+    	return false;
+    }
+    bool Connection::platformCanSendOutgoingMessages() const
+    {
+
+    }
+    bool Connection::sendOutgoingMessage(std::unique_ptr<Encoder> encoder)
+    {
+    	fprintf(stderr,"%s\n",__PRETTY_FUNCTION__);
```

Comment:
A BMessage can be created and encoder can be attached to it by adding parameters like encoder.buffer and encoder.size onto bmessage where we can decode from recieveing end

Reply:
Does Encoder encode the data to a flat buffer necessarily? I thought it would be able to natively fill a BMessage with the high level data (so that PrintToStream on the message does something meaningful, for example).

Well, either way works.